schemas/authoring-guide.md

```diff
@@ -0,0 +1,829 @@
+
+## Named Scalar Types (typedefs)
+
+The non-recursive (scalar) Schema kinds (Boolean, Integer, Float, String, Bytes, Link) may all appear as typedef'd types. That is, a unique name may be assigned to a kind and that name may be used in place of the kind later in the schema. Multiple unique type names may share the same kind.
+
+```ipldsch
+type Foo string
+type Bar int
+type Boom {Foo:Bar}
+```
+
+In terms of data layout, this is equivalent to:
+
+```ipldsch
+type Boom {String:Int}
+```
+
+(Note that even though the Data Model only allows for string keys of maps, the indirection through type `Foo` is allowed since it has a string representation.)
+
+There are a number of reasons to typedef a scalar Schema kind:
+
+* Documentation: A stand-alone type can be more easily documented in the Schema DSL. This may be helpful where there are additional rules that surround a type that are not expressible in the DSL but readers of the Schema may need to be aware of. You will find a lot of such typedefs in the [schema-schema](./schema-schema.ipldsch).
+* Highlighting re-use: Where the re-use of a particular Schema kind is noteworthy, naming it may help in expressing intent.
+* Codegen: the use of named types will have implications for codegen tools. It may be desirable for code generated from a Schema to have recognizable type names in certain positions.
+
+## Links
+
+Links in IPLD Schemas are a special-case. The Data Model kind "Link" is expressed by a token prefixed with the `&` character. The remainder of the token should be `Any` or the name of a type.
+
+Links can be typedef'd, `type Foo &Bar` or can appear inline: `type Baz {String:&Bang}`.
+
+Further, the type name is not a strict assertion that can be directly tested against underlying data, it is simply a hint regarding what should be found when following the link identified by the [CID](../block-layer/CID.md) at the position indicated by the Schema link. Strict assertions of this expected type may be applied at layers above the Schema validation layer when the link is resolved and the node decoded.
+
+For more information about Links in Schemas, see [Links and IPLD Schemas](./links.md).
+
+## Inline Recursive Types
+
+The scalar types (Boolean, Integer, Float, String, Bytes, Link) may appear inline or be typedef'd. In addition, both Map and Link types may appear both inline and as their own type. The additional Schema kinds (Struct, Enum, Union, Copy) do not have an inline variant.
+
+```ipldsch
+type IntList [Int]
+
+type MapOfIntLists {String:IntList}
+
+type Foo {
+  id Int
+  data MapOfIntLists
+}
+```
+
+is equivalent to:
+
+```ipldsch
+type Foo {
+  id Int
+  data [{String:Float}]
+}
+```
+
+As with typedef'd scalar kinds, this has implications for codegen and other API interactions with Schema types. Rather than having an explicit name, `FloatMap`, a generated name may be applied to the List type found at `Foo->data` and the type of the Map nodes found within that list.
```

Comment:
going with this:

> As with typedef'd scalar kinds, this has implications for codegen and other API interactions with Schema types. Rather than having a explicit names, `MapOfIntLists` and `IntList`, auto-generated names may be applied to `Foo->data` and the type of the List nodes found within that Map. (e.g. perhaps `Foo__dataType`, `Foo__data__valueType`).